### DISCUSSION TITLE
Destination Clickhouse - IOException: Error writing request body to server

### DISCUSSION THREAD

**Discussion by Anti0ff** [airbytehq/airbyte#32247] [Connector Questions]
Error when sync data from Postgres to Clickhouse in docker (installed with default params from image: clickhouse/clickhouse-server) with Full refresh

com.clickhouse.client.ClickHouseException: Error writing request body to server, server ClickHouseNode [uri=http://<ip>:8123/default, options={socket_timeout=3000000}
java.io.IOException: Error writing request body to server
java.io.IOException: Broken pipe

22 Gb of data in one table - **10 attempts** in Job History with distinct amount of loaded data... and still not finished, all attempts with this "ClickHouseException"
What problem can be?
Connector is not good working or Clickhouse need some configuration?

Final attempt log:
[1850af91_d5aa_4735_a8cc_dd521bc7052f_job_9_attempt_5_txt.txt](https://github.com/airbytehq/airbyte/files/13282789/1850af91_d5aa_4735_a8cc_dd521bc7052f_job_9_attempt_5_txt.txt)

Clickhouse 23.9.3.12

**Comment by marcosmarxm** [comment]
Are you able to sync a smaller table?
The problem looks the database connector wasn't responsive for some period of time. Maybe lack of resources to handle the writing for a lot of records or somethign similar. Can you check the mem/cpu for the Clickhouse container?

**Reply by Anti0ff** [reply]
1-2 Gb table sync going without problems =\
Assumed that perhaps the reason was a lack of RAM on Clickhouse container in docker, but error message is not very clear
Will try with Clickhouse on separate dedicated server, if there is no real problem in connector

**Reply by blaszta** [reply]
Not sure how much RAM you have, when I load 200+ GB data to ClickHouse on premise server with 16 GB RAM, the server become unresponsive (all other app freeze). Looks like ClickHouse eaten all the RAM (I haven't change the default  setting max_server_memory_usage which is 0 = unlimited).

[a comment by an automated account is not shown]
